core/new-gui/src/app/workspace/component/workflow-editor/workflow-editor.component.ts

```diff
@@ -1,15 +1,151 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, AfterViewInit } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import '../../../common/rxjs-operators';
 
+import * as joint from 'jointjs';
+import { JointUIService } from '../../service/joint-ui/joint-ui.service';
+
+/**
+ * WorkflowEditorComponent is the componenet for the main workflow editor part of the UI.
+ *
+ * This componenet is binded with the JointJS paper. JointJS handles the operations of the main workflow.
+ * The JointJS UI events are wrapped into observables and exposed to other components / services.
+ *
+ * See JointJS documentation for the list of events that can be captured on the JointJS paper view.
+ * https://resources.jointjs.com/docs/jointjs/v2.0/joint.html#dia.Paper.events
+ *
+ * @author Zuozhi Wang
+ * @author Henry Chen
+ *
+*/
 @Component({
   selector: 'texera-workflow-editor',
   templateUrl: './workflow-editor.component.html',
   styleUrls: ['./workflow-editor.component.scss']
 })
-export class WorkflowEditorComponent implements OnInit {
+export class WorkflowEditorComponent implements AfterViewInit {
```

Comment:
Why does this implement AfterViewInit instead of OnInit()

Reply:
In Angular's component lifecycle, afterViewInit() is called after onInit(). 

`onInit` is called after the data of the class are initialized, for example, if this component class needs any inputs from its parent component.

`afterViewInit`, as its name suggests, is called after the view, (the DOM elements) are initialized. We bind JointJS paper to an DOM element, and we need to wait after the view is initialized.